app/models/katello/repository.rb

```diff
@@ -475,11 +477,12 @@ def build_clone(options)
                      :content_id => self.content_id,
                      :content_view_version => to_version,
                      :content_type => self.content_type,
+                     :checksum_type => checksum_type,
+                     :source_repo_checksum_type => source_repo_checksum_type,
```

Comment:
I think this should just be `:checksum_type => checksum_type || source_repo_checksum_type`
 
Meaning that a clone's checksum_type should be explicitly set to whatever checksum type the source repo is using for publishing

Reply:
also to clarify, I don't think we should ever set the :source_repo_checksum_type on a repo unless its after a sync as part of the 'CorrectChecksumType' action

Reply:
> should be explicitly set to whatever checksum type the source repo is using for publishing

So the intent is that if the source repo checksum type changes, the clone will not update its checksum type? I'm not sure I understand why we'd want that.

Reply:
(When i use the term computed_checksum below i'm talking about (checksum_type || source_repo_checksum_type)

so there are two things here: 

a) when creating an archive repo (during publish)
b) when creating or updating a repo in a lifecycle env (during promote)

The cloned created for a)'s checksum should never change.  If a library repo has a sha1 checksum (via the computed checksum) the archive instance created during a publish should also have that checksum and it shouldn't ever change because that 'archive' repo is a 'snapshot' in time.  

for b), the checksum_type of that repo should be updated anytime a new promotion occurs, to that of whatever the checksum type was of the archive repo.